js-foundations.js

```diff
@@ -1,52 +1,152 @@
-// Coding practical questions
-
-// 1. Write an arrow function that takes a number as an argument and decides if the number is evenly divisble by three or not. Use the test cases 15, 0, 385, -7. Expected output: "15 is divisible by three", "0 is divisible by three", "385 is not divisble by three", "-7 is not divisble by three".
-
-
-
-// 2. Create an object called helloMe. Include your firstName, lastName and TWO OTHER properties of you. Add a method to your object that will return a sentence using the object properties to describe you.
-
-
-
-// -------------------------------------------------
-// Consider the variable:
+ ASSESSMENT 2: JAVASCRIPT FOUNDATIONS
+ Coding practical questions
+
+ 1. Write an arrow function that takes a number as an argument and decides if the number is evenly divisble by three or not. Use the test cases 15, 0, 385, -7. Expected output: "15 is divisible by three", "0 is divisible by three", "385 is not divisble by three", "-7 is not divisble by three".
+
+ isDivis = (num) => {
+   if( num%3===0 ){
+     return `${num} is divisible by 3`
+   }else{
+     return `${num} is NOT divisible by 3`
+   }
+ }
+ console.log(isDivis(15));
+ console.log(isDivis(0));
+ console.log(isDivis(385));
+ console.log(isDivis(-7));
+
+ 2. Create an object called helloMe. Include your firstName, lastName and TWO OTHER properties of you. Add a method to your object that will return a sentence using the object properties to describe you.
+
+ let helloMe = {
+   firstName: "Rob",
+   lastName: "Osborn",
+   age: 26,
+   hobby: "tennis",
+   defineMe() {
+     return `Good morning. My name is ${this.firstName} ${this.lastName}. I am ${this.age} years old and, when I'm not coding, I enjoy ${this.hobby}.`
+   }
+ }
+ console.log(helloMe.defineMe());
+ -------------------------------------------------
+ Consider the variable:
 
 var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "temperature", "database"]
 
-// 3a. Write a function that returns every other item as one string. Expected output: "streetlamp teeth nephew database"
-
-
-
-// 3b. Create an arrow function that takes in the variable and returns an array with the number of letters in each string. Expected output: [10, 6, 5, 10, 6, 11, 8]
-
-
-
-// 3c. Stretch: Create a function called capitalizer that takes in the variable and returns the array with all the words capitalized. Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
-
-
-
-// -------------------------------------------------
-// Consider the variables:
+ 3a. Write a function that returns every other item as one string. Expected output: "streetlamp teeth nephew database"
+
+ everyOther = (arr) => {
+   let newArr = []
+   for(let i = 0; i<arr.length; i++){
+     if(i%2===0){
+       newArr.push(arr[i])
+     }
+   }
+   return newArr.join(' ')
+ }
+ console.log(everyOther(randomNouns));
+
+ var everyOther2 = randomNouns.filter(function(num){
```

Comment:
Great Job! Consider using the index argument you have available with _.filter() method.